recipes/faiss/meta.yaml

```diff
@@ -0,0 +1,232 @@
+    'gpu/impl/IVFFlatScan.cuh', 'gpu/impl/IVFPQ.cuh', 'gpu/impl/IVFUtils.cuh',
+    'gpu/impl/L2Norm.cuh', 'gpu/impl/L2Select.cuh', 'gpu/impl/Metrics.cuh',
+    'gpu/impl/PQCodeDistances-inl.cuh', 'gpu/impl/PQCodeDistances.cuh', 'gpu/impl/PQCodeLoad.cuh',
+    'gpu/impl/PQScanMultiPassNoPrecomputed-inl.cuh', 'gpu/impl/PQScanMultiPassNoPrecomputed.cuh',
+    'gpu/impl/PQScanMultiPassPrecomputed.cuh', 'gpu/impl/VectorResidual.cuh',
+    'gpu/utils/DeviceMemory.h', 'gpu/utils/DeviceUtils.h', 'gpu/utils/MemorySpace.h',
+    'gpu/utils/StackDeviceMemory.h', 'gpu/utils/StaticUtils.h', 'gpu/utils/Timer.h',
+    'gpu/utils/BlockSelectKernel.cuh', 'gpu/utils/Comparators.cuh',
+    'gpu/utils/ConversionOperators.cuh', 'gpu/utils/CopyUtils.cuh', 'gpu/utils/DeviceDefs.cuh',
+    'gpu/utils/DeviceTensor-inl.cuh', 'gpu/utils/DeviceTensor.cuh', 'gpu/utils/DeviceVector.cuh',
+    'gpu/utils/Float16.cuh', 'gpu/utils/HostTensor-inl.cuh', 'gpu/utils/HostTensor.cuh',
+    'gpu/utils/Limits.cuh', 'gpu/utils/LoadStoreOperators.cuh', 'gpu/utils/MathOperators.cuh',
+    'gpu/utils/MatrixMult-inl.cuh', 'gpu/utils/MatrixMult.cuh', 'gpu/utils/MergeNetworkBlock.cuh',
+    'gpu/utils/MergeNetworkUtils.cuh', 'gpu/utils/MergeNetworkWarp.cuh',
+    'gpu/utils/NoTypeTensor.cuh', 'gpu/utils/Pair.cuh', 'gpu/utils/PtxUtils.cuh',
+    'gpu/utils/ReductionOperators.cuh', 'gpu/utils/Reductions.cuh', 'gpu/utils/Select.cuh',
+    'gpu/utils/Tensor-inl.cuh', 'gpu/utils/Tensor.cuh', 'gpu/utils/ThrustAllocator.cuh',
+    'gpu/utils/Transpose.cuh', 'gpu/utils/WarpSelectKernel.cuh', 'gpu/utils/WarpShuffles.cuh'
+] %}
+
+package:
+  name: faiss-split
+  version: {{ version }}
+
+source:
+  url: https://github.com/facebookresearch/faiss/archive/v{{ version }}.tar.gz
+  sha256: e1a41c159f0b896975fbb133e0240a233af5c9286c09a28fde6aefff5336e542
+  patches:
+    - patches/0001-change-python-Makefile-to-point-to-libfaiss.-SHLIB_E.patch
+    - patches/0002-remove-fPIC-flag-for-msvc-compat.patch  # [win]
+
+build:
+  number: {{ number }}
+  # TODO: figure out compiler errors on windows
+  skip: true                      # [win]
+  # TODO: remove this to retry building GPU on the feedstock
+  skip: true                      # [cuda_compiler_version != "None"]
+
+requirements:
+  build:
+    - {{ compiler('cxx') }}
+
+outputs:
+  - name: faiss-proc
+    version: 1.0.0
+    build:
+      number: 0
+      string: {{ faiss_proc_type }}
+    test:
+      commands:
+        - exit 0
+    about:
+      home: https://github.com/conda-forge/faiss-feedstock
+      license: MIT
+      license_family: MIT
+      summary: A meta-package to select CPU or GPU build for faiss.
+
+  - name: libfaiss
+    script: build-lib.sh          # [not win]
+    script: build-lib.bat         # [win]
+    build:
+      number: {{ number }}
+      string: "h{{ PKG_HASH }}_{{ number }}_{{ faiss_proc_type }}"
+      run_exports:
+        - libfaiss ={{ version }}=*_{{ faiss_proc_type }}
```

Comment:
Should this be pinned exactly to version?

Reply:
Since I don't know the compatibility guarantees upstream, this seemed like the safest way.

Reply:
Should I reduce it to `version_maj_min`?

Reply:
@mdouze, is faiss using semver?

Reply:
> @mdouze, is faiss using semver?

Yes we aim at sticking to it.

Reply:
@mdouze
Thanks for the quick reply.

@isuruf 
I pinned the run_export to major-version only.

Reply:
Hmmm maybe stay with version_maj_min
We do significant updates in the minor versions even if the API remains compatible. Users should be able to specify 1.6.3 as the version number.

Reply:
@mdouze
I refined the condition and put up a more detailed comment about what it's supposed to achieve. I believe this should match with what you'd like to happen.